mods/ra/maps/sarin-gas-1-crackdown/crackdown.lua

```diff
@@ -42,27 +42,21 @@ SetupTriggers = function()
 		greece.MarkCompletedObjective(objDestroyAllTrucks)
 	end)
 
-	Trigger.OnEnteredFootprint({ TruckEscapeCenter.Location }, function(actor, triggerlose1)
-		if actor.Owner == ussr and actor.Type == "truk" then
-			Trigger.RemoveProximityTrigger(triggerlose1)
-			actor.Destroy()
-			greece.MarkFailedObjective(objDestroyAllTrucks)
-		end
-	end)
-
-	Trigger.OnEnteredFootprint({ EscapeNorth10.Location }, function(actor, triggerlose2)
-		if actor.Owner == ussr and actor.Type == "truk" then
-			Trigger.RemoveProximityTrigger(triggerlose2)
-			actor.Destroy()
-			greece.MarkFailedObjective(objDestroyAllTrucks)
-		end
-	end)
+	SetTrukTrigger(TruckEscapeCenter.Location, triggerlose1)
+	SetTrukTrigger(EscapeNorth10.Location, triggerlose2)
+	SetTrukTrigger(EscapeSouth5.Location, triggerlose3)
+end
 
-	Trigger.OnEnteredFootprint({ EscapeSouth5.Location }, function(actor, triggerlose3)
+SetTrukTrigger = function(location, trigger)
+	Trigger.OnEnteredFootprint({ location }, function(actor, trigger)
 		if actor.Owner == ussr and actor.Type == "truk" then
-			Trigger.RemoveProximityTrigger(triggerlose3)
-			actor.Destroy()
-			greece.MarkFailedObjective(objDestroyAllTrucks)
+			if actor.IsIdle then
+				Trigger.RemoveProximityTrigger(trigger)
+				actor.Destroy()
+				greece.MarkFailedObjective(objDestroyAllTrucks)
+			elseif not actor.IsDead then
+				Trigger.AfterDelay(DateTime.Seconds(0.2), function() SetTrukTrigger(location, trigger) end)
```

Comment:
We are not removing the trigger, so setting it again here looks bogus.

Reply:
By the way: `DateTime.Seconds` only takes integers as argument, so in fact 0.2 will just get rounded down to 0.

Reply:
Okay. I did't find a waitTicks(1) eather. Alternativly we could check for an mine occupying the field as well. I am happy for any other solution : )